mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ExtensionsErrorsDeclarationEnricher.java

```diff
@@ -75,7 +76,7 @@ public WalkDelegateDelegateDeclaration(ExtensionLoadingContext extensionLoadingC
     @Override
     public void onOperation(WithOperationsDeclaration owner, OperationDeclaration operationDeclaration) {
       errorModels.addAll(operationDeclaration.getErrorModels());
-      if (operationDeclaration.isRequiresConnection()) {
+      if (requiresConnectionProvisioning(operationDeclaration)) {
```

Comment:
Changed this one to be based on the SDK flavor instead of the connection provisioning.
Also separated the condition for the `RETRY_EXHAUSTED` error type to be based on the `NoReconnectionStrategyModelProperty`